### PR TITLE
v1.2.0

### PR DESCRIPTION
**Pull request by shun-iwasawa**
Here is my updated to-do list toward releasing the next version:

- [Done] Feature Freeze (Now)
- [Done] Push .TS update PR (Today)
- Merge existing bug-fix PRs
  - [Done] #1604, #1618, #1619, #1621 and #1626
  - [Decided to Postpone] If possible, #1608 
  - ~~Please let me know if there is another PR which you would like to include. Please note in advance some feature PR may be postponed in order to secure sufficient time for review.~~
- [Done] Term for bug-tracking and updating translation ( hopefully until Wednesday, Dec 6 JST )
  - [Done, waiting for feedback] I'm planning to check the mouse and tablet behavior.
- Release V1.2 (on Monday, Dec 11 JST, if no critical issue arises)

### CODE REVIEW

**Comment by manongjohn** [comment]
Can we include #1608 and #1618 ?

**Comment by DoomDesigner** [comment]
What do you need to accept? #1370

**Comment by shun-iwasawa** [comment]
@manongjohn 
I'll include #1618 - sorry, actually I just forgot to include it in the list.
For #1608 , I'll try but I'm a little hesitant including it as it is related to file I/O.
@DoomDesigner 
I haven't reviewed #1370 yet as it is in WIP status.

**Comment by turtletooth** [comment]
@shun-iwasawa Are you willing to look at #1622 and #1602.  They don't affect project data as they are interface tweaks.

**Comment by shun-iwasawa** [comment]
@turtletooth 
For #1622 I would like to ask users in Japanese animation studio the feeling of it. I'm concerning about that the "Apply" button seems to become very small.
For #1602 I'm afraid I don't understand properly the issue yet but wonder if there is a way to adjust the UI size automatically. Please correct me if I'm wrong.
I'm sorry but I didn't include both PRs because - in my brief estimate - they seem to be needed a further discussion.

**Comment by gab3d** [comment]
@shun-iwasawa please ask people at studios, of course. 
It'd be useful also to know their proposals for reducing the footprint of this editor in the best way possible 👍

**Comment by damies13** [comment]
I would like to encourage PR #1141 & PR #1609, with these 2 render farms and local render tasks should now work on all platforms.

**Comment by damies13** [comment]
Also I'd like to ask if the intention for the next release to be 1.2.0 not 1.1.4?

**Comment by shun-iwasawa** [comment]
@damies13
Thank you for tackling with the render farm feature!
I'm terribly sorry that I think they also need some time for verification. I would like them to be included in V1.2.1.

> Also I'd like to ask if the intention for the next release to be 1.2.0 not 1.1.4?

Today I made a release note up until today, listing new features and bug fixes to be introduced in V1.2.0. Thanks to active, motivated and skillful community members, the list became so long (consisted of 157 PRs) that I spent half a day to make it.
Setting the next version to v1.2.0 is because of two reasons.
Firstly, this version is never only be a patch, but also includes fair amount of new features such as timeline UI, MyPaint brush, guided drawing feature, etc. Secondly, by updating minor version number we can separate the stuff folder from the previous version. V1.2.0 will also include many improvement in UI, so separating the stuff folder can avoid expected problems due to the style data being not updated by mistake.
In addition I set the version to v1.2.0 not 2.0.0 is because this version will retain file compatibility with the previous versions in some degree.

**Comment by damies13** [comment]
@shun-iwasawa Thanks for the clarification 👍 

My changes so far has been all about trying to ensure tfarm works everywhere consistently as possible, I have ideas for improving it and adding features (as mentioned in the issues I raised) but I believe it's more important to start from a a state of working everywhere. I also hope that my efforts on tfarm can help people to produce animations more quickly.

**Comment by gab3d** [comment]
@damies13 
Thanks for your work on render farm funcionality!
To us is a very important feature, specially knowing OT is not the fastest program on earth, when it comes to fx and complex composite rendering.

Another related area that would be interesting to revive is the Version Control funtionality that was present in Toonz. 👍

**Comment by psychoanima** [comment]
I am so excited! When 1.2. is going to be released @shun-iwasawa ? :)

**Comment by shun-iwasawa** [comment]
I updated the status.

- Fix for the tablet behavior on OSX was done yesterday (#1659) . Since it is quite a big change, I would like to wait a few days waiting for feedback.
- Due to lack of time for verification, I decided to postpone #1608 to V1.2.1. (Sorry @manongjohn for this. )
- I'm planning to release next version on next Monday, if there will be no critical issue.
- I think small/obvious bug-fixes and translation updates can still be introduced until then.

Please try the current nightly build and check if there are any critical problems, especially related to mouse and tablet behaviors.
Thanks in advance for your cooperation and understanding.

**Comment by RodneyBaker** [comment]
A cursory run through of the latest nightly build is looking good (on
Windows) to include mouse and pen.

**Comment by Rayek** [comment]
@gab3d I would love to see proper GIT support integrated into OpenToonz. Currently only SVN is supported, and I haven't tested it. I use GIT/GitKraken in my projects, although integrated support in OT would be awesome.

**Comment by psychoanima** [comment]
I found only Nightly build that is released on Nov 16, 2016. Am I looking at the wrong download page?

**Comment by RodneyBaker** [comment]
@psychoanima 
That is either a typo in the github code or simply represents when the original posting was made.
The nightly release both on that "main release" page (of which the latest nightly build is currently the 9th release... or the nightly releases on the nightly release page are current even though that date is not.  It looks like it may be plain text and therefore not automatically update.

If we compare the description of that nightly release to that of recent releases listed on the nightly release page you'll see it is indeed a current release*.

https://github.com/opentoonz/opentoonz/releases (9 releases - the nightly release incorrectly describes the date of that release.)

https://github.com/opentoonz/opentoonz_nightlies/releases (529 releases)

* Note that to be sure to get the very latest release we need to look at the actual date for that release because the pages (specifically the nightly release page) sometimes will list them out of order at the top.  This likely has something to do with when it finished compiling rather than when it was submitted for compilation (or something to that effect).  Unless you are specifically testing one new addition to the code you need not be too concerned about which is the latest nightly release but it can trip us up if we are expecting.  A plus (for Window users) would appear to be that the setup file (exe) often won't be listed for those releases that have not fully posted.  Therefore we know we need to look one release down or... simply wait for the next nightly release.

Too many words.  Hope that didn't confuse you.
**Bottom line:  That is the correct nightly release although it may not appear to be**

**Comment by psychoanima** [comment]
thanks @RodneyBaker I was actually downloading it from a wrong page, thanks for the proper link!

**Comment by damies13** [comment]
@Rayek git was one of the things I suggested in issue #1547 before I knew OT had svn support, i've had a brief look at the svn code to see if I could use it as a base for git support, but unfortunately i'm not a guru at either GUI programming or git, so it doesn't look like something I could add quickly. Maybe you could raise a feature request, after iv'e got the toons farm stuff in a good stable place I could have a crack at it, but in the mean time if were lucky someone might beat me to it.

**Comment by caryoscelus** [comment]
@shun-iwasawa #1370 is WIP because not all planned features are implemented, but those that are implemented are working already and should not interfere with the rest of OT.

**Comment by shun-iwasawa** [comment]
@caryoscelus Understood, thank you for clarification.
Since the PR will be automatically closed on merging, IMO it would better to bring the planned feature items out from Pull Requests to Issues (or Google groups). I'm taking `[WIP]` in PR titles as a sign whether the PR is ready to be reviewed or not.

**Comment by shun-iwasawa** [comment]
OK I'll start releasing.

**Comment by masafumi-inoue** [comment]
LGTM
